src/popup/plugins/veeValidate.ts

```diff
@@ -0,0 +1,242 @@
+import { defineRule } from 'vee-validate';
+import { required } from '@vee-validate/rules';
+import BigNumber from 'bignumber.js';
+import { debounce } from 'lodash-es';
+import { Encoding, isAddressValid } from '@aeternity/aepp-sdk';
+import {
+  NETWORK_NAME_MAINNET,
+  NETWORK_NAME_TESTNET,
+  PROTOCOL_AETERNITY,
+} from '@/constants';
+import { isNotFoundError, isUrlValid } from '@/utils';
+import { useBalances, useCurrencies, useAeSdk } from '@/composables';
+import { getAddressByNameEntry, isAensNameValid } from '@/protocols/aeternity/helpers';
+import { isBtcAddressValid } from '@/protocols/bitcoin/helpers';
+import { AE_AENS_DOMAIN, AE_SYMBOL } from '@/protocols/aeternity/config';
+import { tg } from '@/popup/plugins/i18n';
+import { INetwork, ObjectValues, Protocol } from '@/types';
+import { ProtocolAdapterFactory } from '@/lib/ProtocolAdapterFactory';
+import { NameEntry } from '@aeternity/aepp-sdk/es/apis/node';
+
+defineRule(
+  'required',
+  (value: string) => required(value) || tg('validation.required'),
+);
+
+defineRule(
+  'url',
+  (value: string) => !value || isUrlValid(value) || tg('validation.url'),
+);
+
+defineRule(
+  'account',
+  (value: string) => isAddressValid(value) || isAensNameValid(value) || tg('validation.address'),
+);
+
+defineRule(
+  'account_address',
+  (value: string) => isAddressValid(value, Encoding.AccountAddress) || tg('validation.invalidAddress'),
+);
+
+defineRule(
+  'name',
```

Comment:
I suggest that we rename this to `aensName` since we are adding Ethereum support which has its own ENS names

Reply:
Totally make sense. I update the rule name and few others to group them.